src/nplinker/metabolomics/GNPSFileMappingLoader.py

```diff
@@ -0,0 +1,87 @@
+import csv
+from io import TextIOWrapper
+from typing import Dict
+from nplinker.logconfig import LogConfig
+from nplinker.metabolomics.IFileMappingLoader import IFileMappingLoader
+from nplinker.metabolomics.load_gnps import GNPS_FORMAT_NEW_FBMN
+from nplinker.metabolomics.load_gnps import GNPS_FORMAT_OLD_ALLFILES
+from nplinker.metabolomics.load_gnps import identify_gnps_format
+from nplinker.utils import find_delimiter
+
+logger = LogConfig.getLogger(__file__)
+
+FILE_IDENTIFIER_FBMN = " Peak area"
+
+
+class GNPSFileMappingLoader(IFileMappingLoader):
+
+    def __init__(self, filename: str):
+        self._filename: str = filename
+        self._mapping: Dict[int, list[str]] = {}
+        self._gnps_format = identify_gnps_format(filename, False)
+
+        if self._gnps_format is GNPS_FORMAT_OLD_ALLFILES:
+            self.load_mapping_allfiles()
+        elif self._gnps_format is GNPS_FORMAT_NEW_FBMN:
+            self.load_mapping_fbmn()
+        else:
+            raise NotImplementedError(
+                "%{gnps_format} reading not implemented.")
+
+    def mapping(self) -> Dict[int, list[str]]:
+        """Return mapping from spectrum id to files in which this spectrum occurs.
+
+        Returns:
+            Dict[int, list[str]]: Mapping.
```

Comment:
When returned is dict, it's nice to explictly explain what key is and what value is, e.g. for here, it could be `key is spectrum id and value is the spectrum files containing this spectrum`

Reply:
Resolved in df7249d0202902f79931d8c61637c1d85e545941